dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -656,19 +651,7 @@ RtpsUdpDataLink::associated(const RepoId& local_id, const RepoId& remote_id,
       if (rr == readers_.end()) {
         pending_reliable_readers_.erase(local_id);
         RtpsUdpDataLink_rch link(this, OpenDDS::DCPS::inc_count());
-        CORBA::Long an_start = 0;
-        CORBA::Long nf_start = 0;
-        CountMapType::iterator anc_it = acknack_counts_.find(local_id.entityId);
```

Comment:
I thought the purpose of this was to prevent long-running writers from blocking / ignoring short-lived readers that reconnect with the same guid?

Reply:
I'm not sure where the initial nackfrag counter came from but the acknack counter came from StaticDiscovery.  Essentially, a reader can be advertised before it exists so a counter was needed to keep track of the acknack count.  From the code, the nackfrag counter was not used.

The code you are referring to is in `process_acknack` where we check that the count increased but the acknack decreased meaning the reader has gone through rediscovery.  Given the test results, I'll probably have to adjust this code.